paperqa/config.py

```diff
@@ -0,0 +1,413 @@
+        "call {gen_answer_tool_name} tool. The {gen_answer_tool_name} tool output is visible to the user, "
+        "so you do not need to restate the answer and can simply terminate if the answer looks sufficient. "
+        "The current status of evidence/papers/cost is {status}"
+    )
+    paper_directory: str | os.PathLike = Field(
+        default=Path.cwd(),
+        description=(
+            "Local directory which contains the papers to be indexed and searched."
+        ),
+    )
+    index_directory: str | os.PathLike | None = Field(
+        default_factory=lambda: pqa_directory("indexes"),
+        description=(
+            "Directory to store the PQA generated search index, configuration, and answer indexes."
+        ),
+    )
+    manifest_file: str | os.PathLike | None = Field(
+        default=None,
+        description=(
+            "Optional manifest CSV, containing columns which are attributes for a DocDetails object. "
+            "Only 'file_location','doi', and 'title' will be used when indexing."
+        ),
+    )
+    search_count: int = 8
+    wipe_context_on_answer_failure: bool = True
+    timeout: float = Field(
+        default=500.0,
+        description=(
+            "Matches LangChain AgentExecutor.max_execution_time (seconds), the timeout"
+            " on agent execution."
+        ),
+    )
+    should_pre_search: bool = Field(
+        default=False,
+        description="If set to true, run the search tool before invoking agent.",
+    )
+
+    tool_names: set[str] | None = Field(
+        default=None,
+        description=(
+            "Optional override on the tools to provide the agent. Leaving as the"
+            " default of None will use a minimal toolset of the paper search, gather"
+            " evidence, collect cited papers from evidence, and gen answer. If passing tool"
+            " names (non-default route), at least the gen answer tool must be supplied."
+        ),
+    )
+
+    index_concurrency: int = Field(
+        default=30,
+        description="Number of concurrent filesystem reads for indexing",
+    )
+    model_config = ConfigDict(extra="forbid")
+
+    @field_validator("tool_names")
+    @classmethod
+    def validate_tool_names(cls, v: set[str] | None) -> set[str] | None:
+        if v is None:
+            return None
+        # imported here to avoid circular imports
+        from .agents.main import GenerateAnswerTool
+
+        answer_tool_name = GenerateAnswerTool.__fields__["name"].default
+        if answer_tool_name not in v:
+            raise ValueError(
+                f"If using an override, must contain at least the {answer_tool_name}."
+            )
+        return v
+
+
+class Settings(BaseSettings):
+    llm: str = Field(
+        default="gpt-4o-2024-08-06",
+        description="Default LLM for most things, including answers. Should be 'best' LLM",
```

Comment:
If it should be the "best" LLM, shouldn't we use `gpt-4-turbo` here?

Reply:
IMO this is up to the users cost tolerance. So we should also conditions our defaults on being cheaper models.

Reply:
yea - just trying to make it reasonable default for people using the tool without config